Atomic/sign.py

```diff
@@ -50,14 +46,46 @@ def sign(self):
                 _, _, tag = util.decompose(sign_image)
                 tag = ":{}".format(tag) if tag != "" else ":latest"
                 expanded_image_name = str(remote_inspect_info['Name'])
-                sigstore_path = "{}/{}/{}@{}".format(self.args.signature_path, os.path.dirname(expanded_image_name),
-                                                     os.path.basename(expanded_image_name), manifest_hash)
-                self.make_sig_dirs(sigstore_path)
-                sig_name = self.get_sig_name(sigstore_path)
-                fq_sig_path = os.path.join(sigstore_path, sig_name)
-                if os.path.exists(fq_sig_path):
-                    raise ValueError("The signature {} already exists.  If you wish to "
-                                     "overwrite it, please delete this file first")
+
+                if self.args.signature_path:
+                    if not os.path.exists(self.args.signature_path):
+                        raise ValueError("The path {} does not exist".format(self.args.signature_path))
+                    fq_sig_path = os.path.join(self.args.signature_path,
+                                               self.get_sig_name(self.args.signature_path))
+
+                else:
+                    reg, repo, _ = util.decompose(expanded_image_name)
+                    reg_info = util.have_match_registry("{}/{}".format(reg, repo), registry_configs)
+                    # No default and no registry match, exit
+                    if reg_info is None and default_store is None:
```

Comment:
I'm not sure I follow your logic.  The english representation is:

If there is no registry configuration match and no default, then error.